### PR TITLE
telemetry(/dev): add feedback telemetry

### PR DESCRIPTION
**Pull request by chengoramazon**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

`/dev` is missing feedback telemetry. This PR adds the feedback telemetry implementation.
JFeedback

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by willyyhuang — APPROVED** [review]

**Comment by willyyhuang** [comment]
linter failed

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 1]
```diff
@@ -158,6 +164,32 @@ class FeatureDevController(
...
+                comment = objectMapper.writeValueAsString(comment),
+            )
+            logger.info { "$FEATURE_NAME answer feedback sent: \"Negative\"" }
+        } catch (e: Throwable) {
+            e.notifyError(message("feedback.submit_failed", e))
+            logger.warn(e) { "Failed to submit feedback" }
+            return
+        }
+    }
+
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:192]: not happy with how this is a copy paste from the regular chat impl

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 2]
```diff
@@ -771,5 +815,7 @@ class FeatureDevController(
 
     companion object {
         private val logger = getLogger<FeatureDevController>()
+
+        val objectMapper: ObjectMapper = jacksonObjectMapper()
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:819]:
```suggestion

        private val objectMapper = jacksonObjectMapper()
```

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 package software.aws.toolkits.jetbrains.services.amazonqFeatureDev.controller
 
+import com.fasterxml.jackson.databind.ObjectMapper
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:6]:
```suggestion
```

**Review by rli — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt (hunk 1 above)
Comment (chengoramazon) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:191]: Shall I move the common code to a common place as a helper, or modify TelemetryService.sendFeedback to take feature type?

Comment (rli) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:191]: the former makes more sense

**Pull request merged**